libs/core/gasnet_base/include/hpx/gasnet_base/gasnet.hpp

```diff
@@ -0,0 +1,36 @@
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#if (defined(HPX_HAVE_NETWORKING) && defined(HPX_HAVE_PARCELPORT_GASNET)) ||   \
+    defined(HPX_HAVE_MODULE_GASNET_BASE)
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-qual"
+#elif defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
@ctaylor : These `extern "C"` wrappers around the GASNet header include should be unnecessary/extraneous. The GASNet headers already include this declaration internally, and we have several C++ clients that rely upon that.

Did you see a link error prompting you to add these which was resolved by these?